crates/runtimelib/src/messaging/mod.rs

```diff
@@ -128,28 +129,33 @@ impl RawMessage {
             .position(|part| &part[..] == DELIMITER)
             .ok_or_else(|| anyhow!("Missing delimiter"))?;
         let mut parts = multipart.into_vec();
+
         let jparts: Vec<_> = parts.drain(delimiter_index + 2..).collect();
         let expected_hmac = parts.pop().ok_or_else(|| anyhow!("Missing hmac"))?;
         // Remove delimiter, so that what's left is just the identities.
         parts.pop();
         let zmq_identities = parts;
 
+
         let raw_message = RawMessage {
             zmq_identities,
             jparts,
         };
 
-        if let Some(key) = key {
-            let sig = HEXLOWER.decode(&expected_hmac)?;
-            let mut msg = Vec::new();
-            for part in &raw_message.jparts {
-                msg.extend(part);
-            }
-
-            if let Err(err) = hmac::verify(key, msg.as_ref(), sig.as_ref()) {
-                bail!("{}", err);
-            }
-        }
+        // TODO: This seems to be breaking widget messages with binary
```

Comment:
specifically, widgets comm messages with extra buffers:

Reply:
Interesting. It's gotta be on runtimelib's end since widgets is way more battle tested. I didn't work with the binary buffers much so that hasn't gotten use with runtime lib yet (we didn't try with deno right?).

Might be worth checking out the serde setup here as well.